int64 rotation
int64 velocity
---
